src/machines/commerce/__tests__/index.ts

```diff
@@ -0,0 +1,107 @@
+import {rest} from 'msw'
+import {setupServer} from 'msw/node'
+import {interpret} from 'xstate'
+import {SellableResource} from '@types'
+import commerceMachine from '../'
+import {
+  CommerceMachineContext,
+  eggheadPriceCheckUrl,
+  stripeCheckoutSessionUrl,
+} from '../utils'
+const sellableData: SellableResource[] = require('../../../../data/bundles.development.json')
+
+const sellable = sellableData[0]
+
+const defaultContext = {
+  sellable,
+  upgradeFromSellable: null,
+  bulk: false,
+  quantity: 1,
+  stripePriceId: undefined,
+} as CommerceMachineContext
+
+const response200 = [
+  {
+    full_price: 123.45,
+    price: 123.45,
+    slug: 'just-javascript-56c83b7b',
+    url: 'http://app.egghead.af:5000/api/v1/playlists/just-javascript-56c83b7b',
+  },
+]
+const server = setupServer(
+  rest.post(eggheadPriceCheckUrl, (req, res, ctx) => {
+    return res(ctx.json(response200))
+  }),
+  rest.post(stripeCheckoutSessionUrl, (req, res, ctx) => {
+    return res(ctx.json({id: 'test_checkout_session_id'}))
+  }),
+)
```

Comment:
with mock service worker, you can mock specific end points so your code can just run